retry/types.go

```diff
@@ -46,6 +46,10 @@ type ContinueFn func(attempt int) bool
 
 // Retrier is a executor that can retry attempts on executing methods
 type Retrier interface {
+	// SetForeverRetry sets the retry mode that determines whether attempts should
+	// be retried forever.
+	SetForeverRetry(forever bool) Retrier
```

Comment:
Because the server ctor takes in a `Retrier`. I could potentially expose the `Options` method, call `SetForever` on the options, and create a new retrier in `StartForeverAcceptLoop`, although I'll need to store the options in the retrier for that to work.